### PR TITLE
Refactor/codebase updates

### PR DESCRIPTION
**Pull request by anitaschan**
# Description

Added Husky to run ESlint and Prettier on commits. Commits are prevented unless all linting issues are resolved. 

Restructured component folders, there is now a top-level component folder which contains an 'atomic' and a 'composite' folder. 

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Fixed all linting issues bar a couple files (which were added to the gitignore temporarily) as these may require a refactor.

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mobile/.eslintrc.json [hunk 1]
```diff
@@ -14,12 +15,17 @@
   "rules": {
     "jsx-a11y/href-no-hash": ["off"],
     "react/jsx-filename-extension": ["warn", { "extensions": [".js", ".jsx"] }],
+    "react/prop-types": 0,
+    "react/jsx-props-no-spreading": "off",
+    "react/jsx-uses-react": 1,
+    "react/jsx-uses-vars": 1,
```
Comment (adamjamesturner93) [inline mobile/.eslintrc.json:21]: Why is `var` a warning an not an error?

File: mobile/.eslintrc.json [hunk 2]
```diff
@@ -14,12 +15,17 @@
   "rules": {
```
Comment (adamjamesturner93) [inline mobile/.eslintrc.json:15]: can we add `"prettier/prettier": "error"` to make sure we are upholding prettier rules

File: mobile/src/navigation/BottomStackNavigation/BottomStackNavigation.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
+/* eslint-disable react/display-name */
+// TODO: refactor eslint
```
Comment (adamjamesturner93) [inline mobile/src/navigation/BottomStackNavigation/BottomStackNavigation.js:2]: Is this still todo?

Comment (anitaschan) [reply mobile/src/navigation/BottomStackNavigation/BottomStackNavigation.js:2]: nice spot, not in todo anymore, have removed!

File: mobile/.eslintrc.json (hunk 1 above)
Comment (anitaschan) [reply mobile/.eslintrc.json:21]: no particular reason, will change to error

**Review by adamjamesturner93 — APPROVED** [review]

**Pull request merged**